libflux/src/flux/semantic/nodes.rs

```diff
@@ -976,36 +976,44 @@ impl BinaryExpr {
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Comparable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Comparable),
             ]),
             ast::Operator::LessThanOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Comparable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Comparable),
             ]),
             ast::Operator::EqualOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Equatable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Equatable),
             ]),
             ast::Operator::NotEqualOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Equatable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Equatable),
             ]),
             ast::Operator::GreaterThanEqualOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
                 Constraint::Equal(self.typ.clone(), MonoType::Bool),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Equatable),
                 Constraint::Kind(self.left.type_of().clone(), Kind::Comparable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Equatable),
+                Constraint::Kind(self.right.type_of().clone(), Kind::Comparable),
             ]),
             ast::Operator::LessThanEqualOperator => Constraints::from(vec![
-                Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
+                // Constraint::Equal(self.left.type_of().clone(), self.right.type_of().clone()),
```

Comment:
Can you add a link to this issue as part of these comments? https://github.com/influxdata/flux/issues/2393 that way we have the context for why they are commented out for now.